packages/core-sdk/src/resources/ipAccount.ts

```diff
@@ -150,4 +158,41 @@ export class IPAccountClient {
       handleError(error, "Failed to get the token");
     }
   }
+  /**
+   * Sets the metadataURI for an IP asset.
+   */
+  public async setIpMetadata({
+    ipId,
+    metadataURI,
+    metadataHash,
+    txOptions,
+  }: SetIpMetadataRequest): Promise<Hex> {
+    try {
+      const data = encodeFunctionData({
+        abi: coreMetadataModuleAbi,
+        functionName: "setMetadataURI",
+        args: [validateAddress(ipId), metadataURI, metadataHash],
+      });
+      const { txHash } = await this.execute({
+        ipId: ipId,
+        to: coreMetadataModuleAddress[this.chainId],
+        data: data,
+        value: 0,
+        txOptions: {
+          ...txOptions,
+          encodedTxDataOnly: false,
+        },
+      });
+      return txHash!;
+    } catch (error) {
+      handleError(
+        new Error(
+          (error as Error).message
```

Comment:
we should try not recreate the original error unless there's no good alternative. The original error contains more information than just the message. Also the error might not be type `Error` so that needs to be handled as well. Furthermore, this does not retain the original stacktrace, which needs to be fixed.

I'd rather just retain the original error, instead of doing some manual error msg clean up for the `Failed to execute the IP Account transaction:` text.